File: TemplateAPI/TemplateAPI/src/Domain/Entities/AccountSubCategory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TemplateAPI.Domain.Entities;

/// <summary>
///     Account Sub-Category - Second level classification
/// </summary>
public class AccountSubCategory : BaseAuditableEntity
{
    [Required] [MaxLength(50)] public string Code { get; set; } = string.Empty;

    [Required] [MaxLength(100)] public string Name { get; set; } = string.Empty;

    [MaxLength(500)] public string Description { get; set; } = string.Empty;

    public int AccountCategoryId { get; set; }

    [ForeignKey(nameof(AccountCategoryId))]
    public AccountCategory AccountCategory { get; set; } = null!;

    public NormalBalance NormalBalance { get; set; }

    public bool IsActive { get; set; } = true;

    public int DisplayOrder { get; set; }

    // Navigation
    public ICollection<AccountType> AccountTypes { get; set; } = new List<AccountType>();
}

File: angular-template/src/app/modules/shared/models/account-subcategory.ts
export interface AccountSubCategoryDTO {
  id: number;
  code: string;
  name: string;
  description: string;
  accountCategoryId: number;
  accountCategoryName: string;
  normalBalance: string;
  isActive: boolean;
  displayOrder: number;
}

export interface CreateAccountSubCategoryCommand {
  code: string;
  name: string;
  description: string;
  accountCategoryId: number;
  normalBalance: NormalBalance;
  displayOrder: number;
}

export interface UpdateAccountSubCategoryCommand {
  id: number;
  name: string;
  description: string;
  isActive: boolean;
  displayOrder: number;
}

export enum NormalBalance {
  Debit = 1,
  Credit = 2
}
